### ISSUE TITLE
Unresolved reference: PackageList // native_modules.gradle

### ISSUE THREAD

**Issue by sb-bartworx** [react-native-community/cli#600]
## Environment

System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Memory: 26.03 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 28
      Build Tools: 28.0.3
      System Images: android-21 | Google APIs Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom_64
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5692245
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.3 => 0.60.3

## Description

After the upgrade to RN 0.60.3 I can't build my Android App anymore. The build fails with the message: Unresolved reference: PackageList. 

## Reproducible Demo

Just migrate to RN 0.60.3 and the new autolink system.
The PackageList.java file is created from native_modules.gradle (https://github.com/react-native-community/cli/blob/master/packages/platform-android/native_modules.gradle)
It seem that the added (gradle) android.sourceSets.main.java.srcDirs is wrong. 

```gradle
def generatedSrcDir = new File(buildDir, "generated/rncli/src/main/java/com/facebook/react")

task generatePackageList {
    doLast {
      autoModules.generatePackagesFile(generatedCodeDir, generatedFileName, generatedFileContentsTemplate)
    }
  }

preBuild.dependsOn generatePackageList

android {
    sourceSets {
      main {
        java {
          srcDirs += generatedSrcDir
        }
      }
    }
  }
```

The generatedSrcDir contains the (Java) Package. Maybe the issue.
Everything works fine, if I change the script to:

```gradle
def generatedSrcDir = "generated/rncli/src/main/java"
def generatedCodeDir = new File(buildDir, "$generatedSrcDir/com/facebook/react")

task generatePackageList {
    doLast {
      autoModules.generatePackagesFile(generatedCodeDir, generatedFileName, generatedFileContentsTemplate)
    }
  }

preBuild.dependsOn generatePackageList

android {
    sourceSets {
      main {
        java {
          srcDirs += "build/$generatedSrcDir"
        }
      }
    }
  }
```

**Comment by thymikee** [comment]
cc @Salakar

**Comment by Salakar** [comment]
I'm not sure specifically if this is the issue as it currently works as-is on fresh projects, do you have anything custom on your build maybe?

Having said that I've tested your suggested change and can confirm it still builds for me just like before - so if this fixes your project then let's get this change added? Can you send a PR?

I tested both the current code and your change by running `./gradlew clean assembleDebug` in my android directory, both were successfully building.

**Comment by sb-bartworx** [comment]
@Salakar thanks for testing it. You are right. I have some custom changes in my project. I use Kotlin and not Java. I'm sure the problem is related to it. I will test that to confirm that it is really related to Kotlin and then I will send a PR.

**Comment by forsen** [comment]
I usually don't do "+1"s, but since it might strengthen the theory about kotlin being the culprit here. 

We're experiencing the same problem with "Unresolved reference: PackageList", and are also using kotlin for our native code. Our "solution" for the time being was to disable autolinking, but with the suggested changes autolinking started to work.

**Comment by thymikee** [comment]
Can you [update the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version? We've added some extra handling for Kotlin class names.

**Comment by forsen** [comment]
I tried that fix (#563) before it was merged, but it seem that fix is for autolinking to be able to discover modules written in kotlin. It does not fix the issue where `MainApplication.kt` itself is written in kotlin and is unable to resolve `PackageList.java`.

**Comment by thymikee** [comment]
Would you like to send a PR with the fixed proposed by OP?

**Comment by thymikee** [comment]
Published in v2.8.2.

**Comment by sb-bartworx** [comment]
@thymikee @forsen thank for your help. I just added small optimization.
